### PR TITLE
add policy protection to FAA evidences

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/n/projects/2640059/stories/187177668# 

# A brief description of the changes

Current behavior: EvidencesController lacks both pundit policies and spec coverage surrounding it. On top of this, there are several unused or unneeded pieces inside of the controller at this point.

New behavior: Controller cleaned up, specs added (CONTROLLER IN PROGRESS), policies added.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Includes a tweak to the CI/CD pipeline that explicitly groups an existing spec (`auto_extend_income_evidence_spec`) inside the rspec split config file instead of letting the test booster handle it as a leftover spec.  This was necessary because the specs would fail when running on GHA but pass locally.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      def extend_due_date?
+        edit?
+      end
+
+      # Determines if the current user has permission to send out the fdsh hub request.
+      # The user can call fdsh hub if they have permission to perform an edit.
+      #
+      # @return [Boolean] Returns true if the user has permission to call out to the fdsh hub, false otherwise.
+      def fdsh_hub_request?
+        return true if individual_market_admin? #should this leverage hbxadmin policy instead?
```
Comment (ATBull81) [inline components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb:52]: Question for anyone viewing this; should we leverage hbxadmin policy or do a simple check on individual_market_admin here?

File: components/financial_assistance/spec/controllers/financial_assistance/evidences_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+      end
+
+      # this scenario does not seem functionally possible; extend_due_on will return the result from building a new verification history
+      context 'when extending the due date fails' do
+        it 'sets a danger flash message' do
+          # put :extend_due_date, params: params
+          # expect(flash[:danger]).to eq("Unable to extend due date")
+        end
+      end
+    end
```
Comment (ATBull81) [inline components/financial_assistance/spec/controllers/financial_assistance/evidences_controller_spec.rb:157]: Would like an opinion on this. The else block in the evidences_controller#extend_due_date does not look like it should ever really be hit, thus making this check unneeded.

File: components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    def extend_due_date?
+      edit?
+    end
+
+    # Determines if the current user has permission to send out the fdsh hub request.
+    # The user can call fdsh hub if they have permission to perform an edit.
+    #
+    # @return [Boolean] Returns true if the user has permission to call out to the fdsh hub, false otherwise.
+    def fdsh_hub_request?
+      return true if individual_market_admin? #should this leverage hbxadmin policy instead?
```
Comment (ATBull81) [inline components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb:52]: Question for anyone viewing this; should we leverage hbxadmin policy or do a simple check on individual_market_admin here?

Comment (mdkaraman) [reply components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb:52]: The `individual_market_admin?` method actually leverages the permission on the hbx_role. 
See: https://github.com/ideacrew/enroll/blob/6a7806a9589cbe08eeea829b073006594cc6978f/app/policies/application_policy.rb#L160

Specifically, the `modify_family` permission is checked, which is exclusive to the HBX staff role.  So I think the usage of `individual_market_admin?` here is correct.

File: components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      @family ||= record._parent.application.family
+      @applicant ||= record._parent
+    end
+
+    # Determines if the current user has permission to perform an edit.
+    # The user can perform an edit if they are a primary family member in the individual market, an active associated broker in the individual market who has verified their identity, or an admin in the individual market.
+    #
+    # @return [Boolean] Returns true if the user has permission to perform an edit, false otherwise.
+    def edit?
+      FinancialAssistance::ApplicantPolicy.new(user, @applicant).edit?
```
[the remark this answers is not in the document]
Comment (mdkaraman) [reply components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb:28]: I believe we will need the evidence policy for the `fdsh_hub_request?` action, which has a unique permission level.  However, I can see that we don't actually use the `edit?` action in the EvidencesController, so this policy method can be removed.

[the remark this answers is not in the document]
Comment (mdkaraman) [reply components/financial_assistance/app/policies/financial_assistance/evidence_policy.rb:28]: Actually this file can be removed entirely, as the evidence model being used is `Eligibilities::Evidence` and not `FinancialAssistance::Evidence`, which has been deprecated.  We have a pre-existing EvidencePolicy that we will use for authorization.

**Review by harshaellanki — DISMISSED** [review]

**Comment by ATBull81** [comment]
This change (https://github.com/ideacrew/enroll/pull/3673/commits/55f2985aec3e7bbde66e90f5a10b144295391495) was introduced to fix a failing test: when running `components/financial_assistance/spec/domain/financial_assistance/operations/applications/auto_extend_income_evidence_spec.rb` after `spec/controllers/insured/family_members_controller_spec.rb`, there were several failures related to evidences due on dates not being fixed. This will be documented on a ticket where all other leftover files will be moved into the split config.

**Review by mdkaraman — APPROVED** [review]
Looks good!

**Review by polographer — APPROVED** [review]

**Pull request merged**